internal/codespaces/api/api.go

```diff
@@ -392,26 +391,26 @@ func (a *API) GetCodespaceRegionLocation(ctx context.Context) (string, error) {
 	return response.Current, nil
 }
 
-type SKU struct {
+type Machine struct {
 	Name        string `json:"name"`
 	DisplayName string `json:"display_name"`
 }
 
-// GetCodespacesSKUs returns the available SKUs for the user for a given repo, branch and location.
-func (a *API) GetCodespacesSKUs(ctx context.Context, user *User, repository *Repository, branch, location string) ([]*SKU, error) {
-	req, err := http.NewRequest(http.MethodGet, a.githubAPI+"/vscs_internal/user/"+user.Login+"/skus", nil)
+// GetCodespacesMachines returns the codespaces machines for the given repo, branch and location.
+func (a *API) GetCodespacesMachines(ctx context.Context, repoID int, branch, location string) ([]*Machine, error) {
+	reqURL := fmt.Sprintf("%s/repositories/%d/codespaces/machines", a.githubAPI, repoID)
```

Comment:
I suspect this API endpoint is also available under `repos/OWNER/REPO/codespaces/machines`? If so, we can avoid having to resolve a repository to its ID and simply insert the `OWNER/REPO` pair.

Or, if repository ID is absolutely necessary for this route, we could accept a `ghrepo.Interface` type as repository and resolve it to its numeric ID within this method.

Reply:
It's only available under this path at the moment. I like using the repo interface I'll switch to that once it's come out of draft after #4429 lands.

Reply:
For posterity transferring a message I sent to mislav:
>i really like your idea of using ghrepo.Interface in https://github.com/cli/cli/pull/4431 - my goal for the codespace API pkg is to not specify any new types (i have another pr that moves the "codespace" struct out of there) and so I want to move to using ghrepo.Interface based on your suggestion. However, this interface is currently missing a RepoID() method. That's fine I can add it.

It seems that `RepoID() int` is not a method currently supported by `ghrepo.Interface`. Adding this new method will require a few more changes I'd like to keep out of this PR. I will move forward with the current implementation and come back to it when I focus more on the codespaces API pkg.